controller/node_controller.go

```diff
@@ -887,3 +892,47 @@ func (nc *NodeController) createInstanceManager(node *longhorn.Node, imName, ima
 
 	return nc.ds.CreateInstanceManager(instanceManager)
 }
+
+func (nc *NodeController) cleanUpBackingImagesInDisks(node *longhorn.Node) error {
+	settingValue, err := nc.ds.GetSettingAsInt(types.SettingNameBackingImageCleanupWaitInterval)
+	if err != nil {
+		logrus.Errorf("failed to get Setting BackingImageCleanupWaitInterval, won't do cleanup for backing images: %v", err)
+		return nil
+	}
+	waitInterval := time.Duration(settingValue) * time.Minute
+
+	for _, diskStatus := range node.Status.DiskStatus {
+		uuid := diskStatus.DiskUUID
+		backingImages, err := nc.ds.ListBackingImages()
+		if err != nil {
+			return err
+		}
+		for _, bi := range backingImages {
```

Comment:
inverse this so you process the bi's once and have an inner loop for node.Status.DiskStatus per bi.
this way you only update the bi once, after removing all matching disk entries :)